agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import numpy.typing as npt
+from typing_extensions import override
+
+import pinocchio as pin
+
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryPoint
+
+
+class WarmStartReference(WarmStartBase):
+    """
+    A class for generating warmstart values for trajectory optimization problem.
+
+    This class uses a reference trajectory and the robot model to compute the initial state,
+    state vectors, and control inputs.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        # The robot's Pinocchio model, used for forward dynamics computations.
+        self._rmodel: pin.Model | None = None
+        # Data structure associated with the Pinocchio model.
+        self._rdata: pin.Data | None = None
+        # Size of the state vector
+        self._nx: int = 0
+
+    def setup(self, rmodel: pin.Model) -> None:
+        self._rmodel = rmodel
+        self._rdata = self._rmodel.createData()
+        self._nx = self._rmodel.nq + self._rmodel.nv
+
+    @override
```

Comment:
```suggestion
```
I screwed up suggesting it. We need to stick to Python 3.10 which doesn't have this feature yet. It was introduced by [PEP 698](https://peps.python.org/pep-0698/) and targetted towards Python 3.12. Thanks @ArthurH91  for catching this.